MarsSpecFlowProject/MarsSpecFlowProject/StepDefinitions/ThisTestSuiteContainsTestScenariosForLanguageTab_StepDefinitions.cs

```diff
@@ -45,13 +48,176 @@ public void GivenILogIntoThePortalWithUserNameAndPassword(string UserName, strin
         }
 
 
-        
+
+
+        [Given(@"User has no language in their profile")]
+        public void GivenUserHasNoLanguageInTheirProfile()
+        {
+            langobj.DeleteAllElements(driver);
```

Comment:
i see that you are deleting all elements, but this should be put in a place before you run the feature, this way if you run tests in parallel you will not wipe out other test data

Reply:
you also should clean test data after each test 

hence why when you remove data the best thing to do is delete only data you added for the test to be correct
e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test
there for on reset of the two test you should delete english and french for test 1 and hindi for test 2

So to sum up you should delete all data if there is any before the start of any test and then after each test you should delete the data that a particular test is adding!
This is ensure that upon staring the test pack you have reset state and upon finishing each test deletes their data that they added!

also say you are updating a language then you must added language then update it do not assume the language is there!